tutorial/tests/testsuite_helpers.py

```diff
@@ -1,24 +1,27 @@
 import ast
+import html
 import pathlib
 import re
+import traceback
 from dataclasses import dataclass
 from typing import Callable, Dict, List, Optional, Set
 
 import ipywidgets
 import pytest
 from IPython.core.display import HTML, Javascript
 from IPython.display import Code, display
-from nbconvert import filters
 
 
 @dataclass
 class TestResult:
     """Container class to store the test results when we collect them"""
 
-    stdout: str
-    stderr: str
     test_name: str
     success: bool
+    stdout: str | None
+    stderr: str | None
+    exception: List[str] | None
+    traceback: List[str] | None
```

Comment:
I would add a `status` variable instead of `success`. This could have the following states:
- Passed
- Failed
- Syntax error
In this way we can also report syntax errors or any other error that happens during the **test collection** phase but before the test is run.